services/frontend/src/components/profileForms/employmentDataForm/EmploymentDataForm.jsx

```diff
@@ -31,66 +31,31 @@ const EmploymentDataForm = ({ formType }) => {
   const { locale } = useSelector((state) => state.settings);
   const { id } = useSelector((state) => state.user);
 
-  // Get substantive level options
-  const getSubstantiveOptions = useCallback(async () => {
-    const result = await axios.get(`api/option/tenures?language=${locale}`);
-
-    setSubstantiveOptions(result.data);
-  }, [axios, locale]);
-
-  // Get classification options
-  const getClassificationOptions = useCallback(async () => {
-    const result = await axios.get(
-      `api/option/classifications?language=${locale}`
-    );
-
-    setClassificationOptions(result.data);
-  }, [axios, locale]);
-
-  // Get security options
-  const getSecurityOptions = useCallback(async () => {
-    const result = await axios.get(
-      `api/option/securityClearances?language=${locale}`
-    );
-
-    setSecurityOptions(result.data);
-  }, [axios, locale]);
-
-  // Get user profile for form drop down
-  const getProfileInfo = useCallback(async () => {
-    const result = await axios.get(
-      `api/profile/private/${id}?language=${locale}`
-    );
-
-    setProfileInfo(result.data);
-
-    if (result.data.description) {
-      setCharsLeft(1000 - result.data.description.length);
+  const getBackendInfo = useCallback(async () => {
+    try {
+      return await Promise.all([
+        axios.get(`api/option/tenures?language=${locale}`),
+        axios.get(`api/option/classifications?language=${locale}`),
+        axios.get(`api/option/securityClearances?language=${locale}`),
+        axios.get(`api/profile/private/${id}?language=${locale}`),
+      ]);
+    } catch (error) {
+      handleError(error, "redirect", history);
+      throw error;
```

Comment:
In the getBackendInfo from one of the component above, you are setting the state inside of it, should it be the same here?